kiwi/utils/command_capabilities.py

```diff
@@ -58,7 +58,7 @@ def has_option_in_help(
         else:
             arguments = [call] + help_args
         try:
-            command = Command.run(arguments)
+            command = Command.run(arguments, raise_on_error=False)
```

Comment:
mm I am not so sure about that... this method has its own raise_on_error to swallow errors, with the arguable benefit that at least throws a devoted warning message. If one uses this without the raise_on_error it also returns  `KIWICommandCapabilitiesError`. I do not think error filtering is something we will ever need in that context, but who knows...

Moreover I saw you are already calling `has_option_in_help` with `raise_on_error=False`, so I can't see the behavior difference of having this change in this particular case.

Am I missing something?

Reply:
I think you are missing the following

```python
command = Command.run(arguments, raise_on_error=False)
for line in command.output.splitlines():
    if flag in line:
        return True
for line in command.error.splitlines():
    if flag in line:
        return True
```

As you can see we catch information from the error channel of the called command. However this path will never be reached except for commands that prints on stderr but exit with an exit code == 0. Usually commands that prints on stderr also exits with something != 0. In this case Command.run raises with an exception and so all of the second for loop will not be reached. This is also what happens if you call ```setfiles``` it prints its usage and extis with 255. Thus our class cannot be used to fetch data from stderr.

In my opinion this limits the scope a lot and we should not care for the exit code (raise_on_error=False) and allow the class to always see both stdout and stderr data.

So far we used it only in combination with "program --help" which prints on stdout and exits with == 0

Makes sense ?

Reply:
yes you are right, I will adapt the code. Thanks